src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/transport/Codecs.kt

```diff
@@ -294,22 +286,19 @@ object RawMessageCodec : AbstractCodec<RawMessage>(20u) {
 object RawMessageBodyCodec : ByteBufCodec(21u)
 
 object ParsedMessageCodec : AbstractCodec<ParsedMessage>(30u) {
-    override fun read(buffer: ByteBuf): ParsedMessage = ParsedMessage().apply {
+    override fun read(buffer: ByteBuf): ParsedMessage = ParsedMessage.builder().apply {
         buffer.forEachValue { codec ->
             when (codec) {
-                is MessageIdCodec -> id = codec.decode(buffer)
-                is EventIdCodec -> eventId = codec.decode(buffer)
-                is MetadataCodec -> metadata = codec.decode(buffer)
-                is ProtocolCodec -> protocol = codec.decode(buffer)
-                is MessageTypeCodec -> type = codec.decode(buffer)
-                is ParsedMessageRawBodyCodec -> {
-                    rawBody = codec.decode(buffer)
-                    bodySupplier = { buf -> ByteBufInputStream(buf).use { MAPPER.readValue(it) } }
-                }
+                is MessageIdCodec -> setId(codec.decode(buffer))
+                is EventIdCodec -> setEventId(codec.decode(buffer))
+                is MetadataCodec -> setMetadata(codec.decode(buffer))
+                is ProtocolCodec -> setProtocol(codec.decode(buffer))
+                is MessageTypeCodec -> setType(codec.decode(buffer))
+                is ParsedMessageRawBodyCodec -> setRawBody(codec.decode(buffer))
                 else -> println("Skipping unexpected type ${codec.type} value: ${codec.decode(buffer)}")
             }
         }
-    }
+    }.build { buf -> ByteBufInputStream(buf).use { MAPPER.readValue(it) } }
```

Comment:
Either we encapsulate the protocol under the hood or the user will have to deal with it.

Right now, if we are going to keep HTTP in the LwDP there is no reason to use anything different from JSON (because in this case, we will have to convert that protocol into JSON somehow).

Probably, we could add support for different serialization protocols through configuration in the router (or codec) itself. But right now I don't see any new protocols appearing in the nearest future. So, why bother adding something "for the future" if this future might never happen?))

Reply:
Support backward compatibility and migration take a lot of time )

Reply:
Sure, if it will ever happen. Right now there are no cases when we will need something different in the payload. And if we do need support for different serialization protocols, the migration should be done by updating the common library only (because, as I can see it, the decision on which protocol to use will be done under the hood based on the configuration).

And again, even if I add an enum right away it still will have only a JSON protocol. Adding new protocols will mean that we still need to change the code. And common library update will be required

Reply:
Ok, we will faced to changes in both approaches. Let's use current